src/prommis/solid_handling/crusher.py

```diff
@@ -0,0 +1,183 @@
+        )
+        self.properties_out = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            defined_state=True,
+            **self.config.property_package_args,
+        )
+
+        tref = self.flowsheet().time.first()
+        statevars = self.properties_in[tref].define_state_vars()
+
+        for k, v in statevars.items():
+            if k not in ["particle_size_median", "particle_size_width"]:
+                idx = v.index_set()
+                c = Constraint(
+                    self.flowsheet().time,
+                    idx,
+                    doc=f"{k} constraint",
+                    rule=partial(_state_rule, state=k),
+                )
+                self.add_component(k + "_constraint", c)
+
+        # Add Ports
+        self.add_port("inlet", self.properties_in)
+        self.add_port("outlet", self.properties_out)
+
+        self.work = Var(
+            self.flowsheet().time,
+            units=pyunits.W,
+            initialize=3915.17,
+            doc="Work required to increase crush the solid",
+        )
+
+        """ Breakage Distribution calculation as a constraint. 
+        This is the equation for accumulative fraction of solid breakage 
+        probability distribution smaller than size x=feed80size
+        """
+
+        sunit = self.properties_in[tref].particle_size_median.get_units()
+
+        @self.Expression(self.flowsheet().time, doc="Feed P80 Size")
+        def feed_p80(self, t):
+            return (
+                self.properties_in[t].particle_size_median
+                / sunit
+                * (-log(1 - 0.8)) ** (self.properties_in[t].particle_size_width / 2)
+            )
+
+        @self.Expression(self.flowsheet().time, doc="product p80 size")
+        def prod_p80(self, t):
+            return (
+                self.properties_out[t].particle_size_median
+                / sunit
+                * (-log(1 - 0.8)) ** (self.properties_out[t].particle_size_width / 2)
+            )
+
+        @self.Constraint(self.flowsheet().time, doc="Crusher work constraint")
+        def crush_work_eq(self, t):
+            return self.work[t] == (
+                10  # 10 is an empirical correlation, this should not be changed.
+                * self.properties_in[t].flow_mass
+                * self.config.property_package.bond_work_index
+                * (1 / (self.prod_p80[t]) ** 0.5 - 1 / (self.feed_p80[t]) ** 0.5)
+            )
+
+    def _get_stream_table_contents(self, time_point=0):
+        # Dictionary to hold data for all streams
+        io_dict = {"Inlet": self.properties_in, "Outlet": self.properties_out}
+        return create_stream_table_dataframe(io_dict, time_point=time_point)
+
+    def _get_performance_contents(self, time_point=0):
+        # Report
+        var_dict = {
+            "Work Required (W)": self.work[time_point].value,
+        }
+        if hasattr(self, "additional_variable"):
```

Comment:
What is "additional_variable"? I do not see it in your model so there is no need to include this.